azure-chef-extension.gemspec

```diff
@@ -18,7 +18,7 @@ Gem::Specification.new do |s|
   s.executables   = `git ls-files -- bin/*`.split("\n").map{ |f| File.basename(f) }
   s.require_paths = ["lib", "spec"]
 
-  s.add_development_dependency "chef", '<= 16.13.16'
+  s.add_development_dependency "chef"
```

Comment:
You probably still want to pin here

Reply:
This pin was added when we were pinning the azure chef extension to Chef-Infra-Client 16, so to removing the pin for latest Chef Infra Client this pin will need to removed from here.